### PR TITLE
Chronos: spark df support in XShardsTSDataset

### PR DESCRIPTION
**Pull request by TheaperDeng**
related to #4472 

Basic usage:
```python
df = <pyspark.sql.dataframe.DataFrame> # created by any methods
tsdataset = XShardsTSDataset.from_sparkdf(df, ...)
tsdataset.roll(lookback=..., horizon=...)
data = tsdataset.to_xshards()

f = Forecaster(...)
f.fit(data, ...)
```

### CODE REVIEW

**Comment by TheaperDeng** [comment]
http://10.112.231.51:18888/job/BigDL-PRVN-chronos-Python-Spark-2.4-py36-ray-part2/727/
http://10.112.231.51:18888/job/BigDL-Chronos-PR-Validation/739/
http://10.112.231.51:18888/job/BigDL-Orca-PR-Validation/630/

File: python/dllib/src/bigdl/dllib/utils/utils.py [hunk 1]
```diff
@@ -212,7 +216,7 @@ def convert_for_cols(row, cols):
             elif isinstance(row[name], DenseVector):
                 result.append(row[name].values.astype(np.float32))
             else:
-                invalidInputError(isinstance(row[name], SparseVector),
+                invalidInputError(not isinstance(row[name], SparseVector),
```
Comment (yushan111) [inline python/dllib/src/bigdl/dllib/utils/utils.py:219]: Maybe we shouldn't add `not` here...

File: python/chronos/test/bigdl/chronos/data/experimental/test_xshardstsdataset.py [hunk 2]
```diff
@@ -181,3 +191,16 @@ def test_xshardstsdataset_roll_multiple_id(self):
         collected_numpy = shards_numpy.collect()  # collect and valid
         x = np.concatenate([collected_numpy[i]['x'] for i in range(len(collected_numpy))], axis=0)
         assert x.shape == ((50-lookback-horizon+1)*2, lookback, 2)
+
+    def test_xshardstsdataset_sparkdf(self):
+        df = generate_spark_df()
+        tsdata = XShardsTSDataset.from_sparkdf(df, dt_col="date",
+                                               target_col="feature",
+                                               id_col="id")
```
Comment (yushan111) [inline python/chronos/test/bigdl/chronos/data/experimental/test_xshardstsdataset.py:199]: Shall we also add test case for user passing `id_col = None`?

File: python/orca/src/bigdl/orca/learn/utils.py [hunk 3]
```diff
@@ -291,6 +316,22 @@ def _dataframe_to_xshards(data, feature_cols, label_cols=None, accept_str_col=Fa
     return SparkXShards(shard_rdd)
 
 
+def _dataframe_to_xshards_of_pandas_df(data, feature_cols, label_cols=None, accept_str_col=False):
```
Comment (yushan111) [inline python/orca/src/bigdl/orca/learn/utils.py:319]: Maybe we could make the function public

Comment (yushan111) [reply python/orca/src/bigdl/orca/learn/utils.py:319]: Shall we also add some docs to explain the `data` as well as `feature_cols`/`label_cols` constraints?  e.g one feature_col corresponds to 1-d data

File: python/dllib/src/bigdl/dllib/utils/utils.py (hunk 1 above)
Comment (TheaperDeng) [reply python/dllib/src/bigdl/dllib/utils/utils.py:219]: This is a comment sent by @cyita

Comment (yushan111) [reply python/dllib/src/bigdl/dllib/utils/utils.py:219]: It seems to be irrelevant to the general purpose of this PR.  We could raise a new PR to fix this (if needed). 

Besides, to fix this, we may need to add some extra logic for converting a `SparseVector` to numpy ndarray, and then raise error for other input cases.

Comment (TheaperDeng) [reply python/dllib/src/bigdl/dllib/utils/utils.py:219]: sure, I will split this for another PR.

**Comment by TheaperDeng** [comment]
http://10.112.231.51:18888/job/BigDL-Chronos-PR-Validation/755/
http://10.112.231.51:18888/job/BigDL-PRVN-ray-ctx-example/86/
http://10.112.231.51:18888/job/ZOO-PR-BigDL-Python-Spark-3.1-py37/663/
http://10.112.231.51:18888/job/BigDL-Orca-PR-Validation/633/

**Review by yushan111 — APPROVED** [review]
